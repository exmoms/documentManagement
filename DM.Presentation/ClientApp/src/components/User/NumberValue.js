import React, { Component } from "react";
import TextField from "@material-ui/core/TextField";
import { withTranslation } from "react-i18next";

class NumberValue extends Component {
  constructor(props) {
    super(props);
    this.state = { value: 0 };
    this.handleValueChange = this.props.handleValueChange;
  }

  t = this.props["t"];

  handleChange = (event) => {
    let value = parseInt(event.target.value);
    this.setState(
      { value: value },
      this.handleValueChange(value, this.props.label)
    );
  };

  render() {
    return (
      <div>
        <TextField
          id={this.props.label}
          label={this.t(this.props.label)}
          type="number"
          onChange={this.handleChange}
          InputLabelProps={{
            shrink: true,
          }}
        />
      </div>
    );
  }
}

export default withTranslation()(NumberValue);
