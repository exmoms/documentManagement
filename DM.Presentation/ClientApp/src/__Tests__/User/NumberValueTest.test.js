import React from "react";
import ReactDom from "react-dom";
import NumberValue from "../../components/User/NumberValue";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";
import { mount } from "enzyme";
import TextField from "@material-ui/core/TextField";
afterEach(cleanup);
test("NumberValue matches snapshot", () => {
  const component = renderer.create(<NumberValue />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

describe("NumberValue", () => {
  test("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDom.render(<NumberValue></NumberValue>, div);
  });

  it("NumberValue Functionality ", () => {
    let labelMock = "Text Value";
    let handleValueChangeMock = jest.fn();
    let attributeNameMock = "attrib1";
    let wrapper = mount(
      <NumberValue
        label={labelMock}
        handleValueChange={handleValueChangeMock}
        attributeName={attributeNameMock}
      />
    );
    act(() => {
      wrapper
        .find(TextField)
        .at(0)
        .props()
        .onChange({ target: { value: 5 } });
    });

    wrapper.update();

    expect(handleValueChangeMock).toHaveBeenCalledTimes(1);
    expect(handleValueChangeMock.mock.calls[0][0]).toStrictEqual(5);
    expect(handleValueChangeMock.mock.calls[0][1]).toStrictEqual(labelMock);
  });
});
