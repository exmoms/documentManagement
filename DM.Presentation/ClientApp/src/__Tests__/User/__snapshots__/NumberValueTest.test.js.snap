// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberValue matches snapshot 1`] = `
<div>
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="number"
      />
    </div>
  </div>
</div>
`;
